obelix/panoramix/services/ethereum/client.go

```diff
@@ -0,0 +1,127 @@
+package ethereum
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"github.com/Cleopha/ifttt-like-common/credentials"
+	"github.com/Cleopha/ifttt-like-common/protos"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/structpb"
+	"math/big"
+	"os"
+)
+
+var (
+	ErrFailedToConvertPublicKey = errors.New("failed to convert public key to valid format")
+)
+
+type Client struct {
+	ctx        context.Context
+	clt        *ethclient.Client
+	privateKey *ecdsa.PrivateKey
+}
+
+func NewClient(ctx context.Context) *Client {
+	return &Client{
+		ctx: ctx,
+	}
+}
+
+func (c *Client) configure(owner string) error {
+	clt, err := ethclient.Dial(os.Getenv("INFURA_ENDPOINT"))
+	if err != nil {
+		return fmt.Errorf("failed to create ethereum client: %w", err)
+	}
+
+	c.clt = clt
+
+	credentialsClient, err := credentials.NewClient(os.Getenv("CREDENTIAL_API_PORT"))
```

Comment:
pls in init and check if CREDENTIAL_API_PORT is set

Reply:
Fixed in https://github.com/Cleopha/ifttt-like/pull/76/commits/485f0b1571ae6d34ece509592cc9a4cde32bc17b